src/test/java/domain/LadderGameTest.java

```diff
@@ -5,36 +5,124 @@
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.MethodSource;
 
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
 
 public class LadderGameTest {
 
     private LadderGame ladderGame;
 
     @BeforeEach
     void before() {
-        List<String> playerNames = List.of("a", "b", "c");
+        List<String> playerNames = List.of("a", "b", "c", "d", "e");
         Players players = PlayersFactory.of(playerNames);
-        int ladderHeight = 10;
-        Ladder ladder = LadderFactory.of(playerNames.size(), ladderHeight, new RandomBasedStrategy());
-        ladderGame = new LadderGame(players, ladder);
+        GameResults gameResults = new GameResults(
+                5,
+                List.of(new GameResult("당첨"), new GameResult("꽝"), new GameResult("당첨"), new GameResult("꽝"), new GameResult("꽝"))
+        );
+        List<Point> pointsOfFirstLine = List.of(Point.EXIST, Point.NOT_EXIST, Point.NOT_EXIST, Point.EXIST);
+        List<Point> pointsOfSecondLine = List.of(Point.NOT_EXIST, Point.EXIST, Point.NOT_EXIST, Point.EXIST);
+        List<Point> pointsOfThirdLine = List.of(Point.NOT_EXIST, Point.EXIST, Point.NOT_EXIST, Point.NOT_EXIST);
+        List<Point> pointsOfFourthLine = List.of(Point.EXIST, Point.NOT_EXIST, Point.EXIST, Point.NOT_EXIST);
+        CompleteBasedStrategy completeBasedStrategy = new CompleteBasedStrategy(
+                List.of(
+                        pointsOfFirstLine, pointsOfSecondLine, pointsOfThirdLine, pointsOfFourthLine
+                )
+        );
+        int ladderHeight = 4;
+        Ladder ladder = LadderFactory.of(playerNames.size(), ladderHeight, completeBasedStrategy);
```

Comment:
```suggestion
        Line pointsOfFirstLine = new Line(List.of(Point.EXIST, Point.NOT_EXIST, Point.NOT_EXIST, Point.EXIST));
        Line pointsOfSecondLine = new Line(List.of(Point.NOT_EXIST, Point.EXIST, Point.NOT_EXIST, Point.EXIST));
        Line pointsOfThirdLine = new Line(List.of(Point.NOT_EXIST, Point.EXIST, Point.NOT_EXIST, Point.NOT_EXIST));
        Line pointsOfFourthLine = new Line(List.of(Point.EXIST, Point.NOT_EXIST, Point.EXIST, Point.NOT_EXIST));

        Ladder ladder = new Ladder(List.of(
                pointsOfFirstLine, pointsOfSecondLine, pointsOfThirdLine, pointsOfFourthLine
        ));
```
현재 코드와 위 코드는 어떤 차이점이 있을까요?

Reply:
Ladder를 더 효율적으로 생성하기 위해 만든 LadderFactory 때문에 오히려 불필요한 코드가 늘어났다는 생각을 했습니다.
그렇다면 LadderFactory는 불필요한 클래스인가에 대해서도 고민해보았습니다.

먼저 팩토리 패턴을 공부하면서 LadderFactory가 올바르게 팩토리 패턴을 적용하여 사용되고 있는지 살펴보았는데요,
제가 작성한 LadderFactory는 팩토리 패턴의 장점을 잘 활용했다고 보기엔 어렵다는 생각이 들었습니다.
공부한 팩토리 패턴의 장점은 다양한 유형의 객체를 코드의 중복 또는 변경 없이 생성할 수 있다는 것이기 때문입니다.

그런데 LadderFactory가 없다면 Ladder를 만드는 책임을 누가 가져갈 것인가에 대해서 고민해보면서 Ladder를 입력 받은 크기의 Ladder 객체로 만들기 위한 책임을 갖는 클래스는 여전히 필요하다고 생각했습니다.
따라서 우선 LadderFactory는 남겨두고, 테스트 코드에서 사용하기 위해 만든 CompleteBasedStrategy는 삭제하였습니다. 그리고 Ladder는 생성자로 직접 생성해주는 방향으로 코드를 수정하였습니다.

이 부분은 제가 짠 코드에 아직 확신이 서지 않아 팩토리 패턴에 대해서 조금 더 공부를 해야할 것 같습니다ㅠㅠ

Reply:
자동차 경주 미션에서 random에 대한 테스트를 배운 뒤, 다음 미션에서 random이 나오면 대부분 랜덤을 주입하려고 하더라구요. 

현재 LineFactoryTest에 있는 연속되지 않는 point에 대한 검증은 Line에서 하고 있습니다. 그렇다면 LadderFactory, LineFacotry를 테스트하는 건 어떤 의미가 있을까요? 
단순히 런타임에 에러가 나지 않는다라는 걸 얘기하는 용도로 테스트를 작성할 수는 있을 것 같아요. 이렇게 생각하면 처음부터 random 값에 대한 생성을 주입 등을 이용해서 조절할 필요가 없을 것 같아요.

오히려 LineFactory 등이 존재해야할 이유는 random 값을 자체로 생성할 수 있는 데서 찾아야하는 게 아닐까 하는 생각도 드네요. 

여튼, 팩토리 클래스를 사용하든 팩토리 메서드를 사용하든 관습적으로 만들기보다는 현재 요구사항을 구현하기 위해서 어떻게 구현할지를 고민해보시면 좋을 것 같네요 😄

Reply:
활용성을 고민하기보다는 습관적으로 팩토리 패턴을 사용하려고 했던 것 같습니다. 

코멘트 남겨주신 것처럼 Line을 생성할 때 연속된 Point에 대한 유효성 검증을 진행하기 때문에 테스트 코드를 위한 Factory 패턴이 되었던 것 같다는 생각이 들어 LineFactory에서 직접 random 값을 생성하도록 리팩토링 진행하였습니다.
프로그램의 요구사항에 따라 적절한 방식을 사용해보려고 노력해보겠습니다!